website/sampling-consensus.md

```diff
@@ -1,37 +1,79 @@
+It’s technically possible to run LLM inference as a smart contract on-chain, but it would incur huge costs. Therefore we have to look into alternative methods of verification. 
 
-This is especially relevant to enforce verifiable compute. Indeed, by enforcing determinism across processed requests, we can rely on the following observation:
+# Alternative methods of consensus 
 
-`Observation:` Through the Atoma sampling consensus protocol, whenever two or more nodes are selected to execute a request, the select nodes disagree if at least one of the node has been dishonest. In particular, a honest node will never agree on the state of a request output with dishonest one.
+##  Byzantine Fault Tolerance
 
-## Probabilistic considerations for Atoma's Sampling Consensus
+Byzantine Fault Tolerance (BFT) consensus is a method used in distributed systems to agree on a single data value or decision even if a proportion of the nodes in the system are unreliable or malicious. BFT consensus ensures that even if some participants are unreliable, the system as a whole can still make a reliable and unanimous decision.
 
-As previously mentioned, the Atoma protocol can enforce determinism for
-execution workloads (a.k.a requests) by compute replication, through selecting uniformly at random a given number of nodes to run the same compute. 
+The issue with BFT consensus is that the large majority of nodes (usually around 2/3rds) must agree on a transaction or block of transactions.
 
-Depending on the number of sampled nodes, different levels of output integrity are reached. For example, assuming a dishonest participant of the
-Atoma Network controls a percentage `r` of the network, the probability, `P`, that a quorum of `N > 0` selected nodes (including at least one of the dishonest participant nodes) is
+In the context of AI inference, BFT methods require heavy compute distributed across many nodes with each node performing the same computations independently. This incurs problems with computational power, latency, and scalability.
 
-`P = r^N`.
+##  Zero Knowledge Machine Learning 
 
-In particular, if `N = 5` nodes are selected and the dishonest participant controls one third of the network, that is `r = 1/3`, the probability above becomes
+Zero knowledge machine learning (ZKML) is a technique to ensure data remains private through zero knowledge proofs (ZKPs). When training the LLM, computations are completed without revealing any of the information provided. 
```

Comment:
Data doesn't remain private with ZKML, only potentially the model weights

Reply:
Why training, shouldn't it be inference ?

Reply:
Changed to inference, still unsure as to exactly what to write. Please review